# Larissa Kidd 9/12/2023
## Instragram Likes Report
 I hypothesize that "pubity" and "memezar" are in my top 20 accounts liked  because I have followed them for a long time and I like pubity's news reports and I enjoy memes

In [2]:
import pandas as pd
import json

In [3]:
with open('/Users/larissakidd/Downloads/Fourth Year/Fall 23/Data in EMAT/instagram/likes/liked_posts.json') as j:
    my_data = json.load(j)

### I now have my instagram likes loaded
Now I want to create a dataframe from them

In [4]:
my_df = pd.DataFrame(my_data['likes_media_likes'])

### I want to add timestamps - so I need to find it in the data

In [5]:
my_data['likes_media_likes'][0]['string_list_data'][0]['timestamp']

1693581754

### now to add these into a variable

In [6]:
time = [x['string_list_data'][0]['timestamp'] for x in my_data['likes_media_likes']] 
time[0:10]

[1693581754,
 1693581074,
 1693581036,
 1693515116,
 1693515083,
 1693493787,
 1693493777,
 1693447022,
 1693428135,
 1693417571]

### Now that I found it, I need to add time as a column in my data

In [7]:
my_df['likes_timestamps'] = time
my_df[0:11]

title  \
0                    pubity   
1            kentsportswear   
2             larochesoccer   
3                   memezar   
4         hannah.lohmueller   
5                 _abbydev_   
6            rachelostavitz   
7      kentstatepanhellenic   
8   kellysummersphotography   
9           christina.black   
10                pnwdad206   

                                     string_list_data  likes_timestamps  
0   [{'href': 'https://www.instagram.com/p/CwpdJOQ...        1693581754  
1   [{'href': 'https://www.instagram.com/p/CwpZ71s...        1693581074  
2   [{'href': 'https://www.instagram.com/p/CwpxFoC...        1693581036  
3   [{'href': 'https://www.instagram.com/p/CwnxypU...        1693515116  
4   [{'href': 'https://www.instagram.com/p/CwnUu6z...        1693515083  
5   [{'href': 'https://www.instagram.com/p/Cwm_eNT...        1693493787  
6   [{'href': 'https://www.instagram.com/p/CwksJAI...        1693493777  
7   [{'href': 'https://www.instagram.com/p/CwiPqF5...        1693447022  
8   [{'href': 'https://www.instagram.com/p/CwlB5Jf...        1693428135  
9   [{'href': 'https://www.instagram.com/p/Cwf9yJQ...        1693417571  
10  [{'href': 'https://www.instagram.com/reel/Cv7b...        1693417490

In [8]:
grouped = my_df.groupby('title').count()
grouped.shape

(1382, 2)

### While I only show 10 accounts (titles), there are 1,382 accounts in the timestamp dataframe

Now, I want to know quantity of likes over time so I need to count them!

In [9]:
grouped.sort_values('likes_timestamps', ascending = False).head(20)

string_list_data  likes_timestamps
title                                                      
cohmedy                               199               199
hhigley00                             144               144
lilygcashmoneyap                      144               144
ksudeltagamma                         137               137
pubity                                120               120
zk_k18                                108               108
fab_ksu                               104               104
lilygracepittman                      100               100
xo.aubs_                               81                81
daquan                                 78                78
kellysummersphotography                74                74
barstoolkent                           64                64
mat_thew_please                        58                58
courtney_sayre                         55                55
theanairene                            53                53
tampon                                 53                53
ksudgphilanthropy                      49                49
jenicakidd                             48                48
scoobydoofruitsnacks                   48                48
shockran99                             46                46

### Contrary to my hypothesis, memezar is not in the top 20.
To settle my curiosity, I need to find it where it lies.

In [10]:
grouped.loc['memezar']

string_list_data    34
likes_timestamps    34
Name: memezar, dtype: int64

## While "pubity" is in the top 5 with 120 likes,  "memezar" is well below the top 20 with only 34 likes. 
Upon seeing the results of my top-liked accounts, I also realized that I neglected to think about my long time best friend, my sorority, and my boyfriend which are also obviously in the top 5... whoops

## If I could reapproach:
### I would figure out the timeline of how far back this data goes 
It is hard to read the timestamps because of their format so I might have misjudged based on how far back this data is collected from. To mitigate this problem, I would have to convert the time into something legible in regularly used time/date format.  

## Limitations
I was surprised with how well this data worked and I felt like it was relatively easy to manipulate. The only issue was finding how far back this data went as far as the timeline and it was difficult to find where I had look to get the data I needed because they are deeply imbedded. 